README.md

```diff
@@ -2,70 +2,52 @@
-needs to be managed by the framework is added. Spring is not aware of objects
-unless it’s explicitly told to keep track of them. We can also define
-relationships between objects in the context.
+Now that we have a very simple working MVC, let's back up a bit and look
+at some more fundamentals of the Spring framework.
 
-Beans are object instances that are managed by Spring. We can add beans using in
-the context by using the `@Bean` annotation or stereotype annotations provided
-by Spring.
-
-## Create Maven Project
-
-We will be using Maven to add the Spring context dependency to our project.
+In this lesson, we'll talk about what an application context is and how we can
+add objects to the Spring application context.
 
-1. Open IntelliJ and create a new project. Name the project
-   `spring-context-example`.
+## What is the Spring Context?
 
-   ![IntelliJ create project screen](https://curriculum-content.s3.amazonaws.com/java-spring-1/create-project-spring-context.png)
+A fundamental element to the Spring framework is the **application context**. The
+context is a memory space in our application in which we can add objects to that
+need to be managed by the framework. Spring is not aware of objects unless it's
+explicitly told to keep track of them. We can also define relationships between
+objects in the context. This is an important concept to using Spring, because
+otherwise, we cannot turn our Java objects into something that Spring recognizes.
 
-2. Add the `spring-context` dependency in the `pom.xml` file and reload the
-   project.
+So, how do we tell Spring to add our Java objects into the Spring application
+context?
 
-   ```xml
-   <dependencies>
-       <dependency>
-           <groupId>org.springframework</groupId>
-           <artifactId>spring-context</artifactId>
-           <version>5.3.15</version>
-       </dependency>
-   </dependencies>
-   ```
-
-3. Create a `main` and `config` package in `src/main/java`. Your directory
-   structure should look like this:
-
-   ```
-   ├── pom.xml
-   └── src
-       ├── main
-       │   ├── java
-       │   │   ├── config
-       │   │   └── main
-       │   └── resources
-       └── test
-           └── java
-   ```
+This is where we will introduce the concept of a bean and a component.
 
 ## Add Bean using `@Bean` Annotation
 
+**Beans** are object instances that are managed by Spring. We can add beans into
+the context by using the `@Bean` annotation or stereotype annotations provided
+by Spring.
+
 The `@Bean` annotation is applied to methods. We will create a `Dog` class and
 add an instance of it to the Spring context.
 
 ![Spring context diagram with a single dog bean](https://curriculum-content.s3.amazonaws.com/java-spring-1/spring-context-dog.png)
 
-Let’s create the `Dog` class before we configuring Spring and adding the bean.
-Create the `Dog` class in the `main` and add the following code:
+Let’s create a `Dog` class before we configure Spring and add the bean to the
```

Comment:
Maybe we should consider a class that's closer to what the students will see on the job.
At my last job, we used a lot of RestTemplates beans. Perhaps that could work?
Or maybe a simpler example than RestTemplate would be a class that does some kind of String formatting?

Reply:
I definitely want to show a RestTemplate bean in a later lesson. I wasn't sure if introducing that here would be too much / take away from the context (no pun intended)? But I do like the string formatting idea. Let me think on this and I'll try to refactor it.

Reply:
@developerjaym - Do you mind if I come back to this lesson if I have time after reworking Spring Mod 2 and I push this up as is for now?